src/SourceCode.Chasm/TreeNodeMap.cs

```diff
@@ -282,70 +333,73 @@ private static ReadOnlyMemory<TreeNode> DistinctSort(TreeNode[] array, bool alre
-                        if (StringComparer.Ordinal.Equals(array[0].Name, array[1].Name))
-                        {
-                            // If it's a complete duplicate, silently skip
-                            if (TreeNodeComparer.Default.Equals(array[0], array[1]))
-                                return new TreeNode[1] { array[0] };
-
-                            // Else throw
+                        if (TreeNodeComparer.Default.Equals(array[0], array[1]))
+                            result = new TreeNode[1] { array[0] };
+                        else
                             throw CreateDuplicateException(array[0]);
-                        }
-
-                        // Sort
-                        if (TreeNodeComparer.Default.Compare(array[0], array[1]) < 0)
-                            return new TreeNode[2] { array[0], array[1] };
-
-                        return new TreeNode[2] { array[1], array[0] };
                     }
+                    // Sort
+                    else if (TreeNodeComparer.Default.Compare(array[0], array[1]) > 0)
+                    {
+                        result = new TreeNode[2] { array[1], array[0] };
+                    }
+                    else if (alreadyCopied)
+                    {
+                        result = array;
+                    }
+                    else
+                    {
+                        result = new TreeNode[2] { array[0], array[1] };
+                    }
+                    return result;
 
                 default:
+
+                    // If callsite did not already copy, do so before mutating
+                    var nodes = array;
+                    if (!alreadyCopied)
                     {
-                        // If callsite did not already copy, do so before mutating
-                        var nodes = array;
-                        if (!alreadyCopied)
-                        {
-                            nodes = new TreeNode[array.Length];
-                            array.CopyTo(nodes, 0);
-                        }
+                        nodes = new TreeNode[array.Length];
+                        array.CopyTo(nodes, 0);
+                    }
 
-                        // Sort: Delegate dispatch faster than interface (https://github.com/dotnet/coreclr/pull/8504)
-                        Array.Sort(nodes, TreeNodeComparer.Default.Compare);
+                    // Sort: Delegate dispatch faster than interface (https://github.com/dotnet/coreclr/pull/8504)
+                    Array.Sort(nodes, TreeNodeComparer.Default.Compare);
 
-                        // Distinct
-                        var j = 1;
-                        for (var i = 1; i < nodes.Length; i++)
+                    // Distinct
+                    var j = 1;
+                    for (var i = 1; i < nodes.Length; i++)
+                    {
+                        // If the Name (alone) is duplicated
```

Comment:
This algorithm should match the other one where I commented.
And checking the Name then the Whole is correct

Reply:
The only change on these lines is whitespace.

Reply:
Weird. Even when I use 'View', it shows up as different